robot_sdk/SDK/owl_ec_sdk/robot.py

```
@@ -570,12 +633,18 @@ def getRobotMode(self) -> str:
             "OPERATING",
             "REMOTE"
         ]
-
-        if self.connect_success:
-            _, result = self.execute("getRobotMode")
+        self.connect()
+        try:
+            # Check if the robot is connected
+            if not self.connect_success:
+                raise Exception(f"Robot not connected at IP:{self.ip_address}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)